paytrail/views.py

```diff
@@ -292,9 +294,47 @@ def render_payment_details(self, request, **kwargs):
 
 
 def notification(request, token):
-    """Paytrail server calls this when payment was successful."""
+    """
+    Paytrail server calls this when payment was successful.
+    This can be called multiple times per one payment.
+    Creates a transaction object if not already created.
+    """
     validate_signature(request.GET)
-    get_model('order', 'Order').objects.get(
-        number=signer.unsign(token)
-    ).sources.get(source_type=get_source_type()).debit()
+    order = get_object_or_404(Order, number=signer.unsign(token))
+
+    transaction_id = request.GET.get('checkout-transaction-id', None)
+    if not transaction_id:
+        logger.error('checkout-transaction-id missing', code=400)
+        raise ValidationError
+
+    status = request.GET.get('checkout-status', None)
+    if not status:
+        logger.error('checkout-status query parameter missing')
+        raise ValidationError('checkout-status missing', code=400)
+
+    if status in ['ok', 'pending', 'delayed']:
+        # Payment source is created in SuccessView.
+        # It is possible that this callback view is called before SuccessView
+        # so source might not be available yet.
+        source = get_object_or_404(order.sources.all(), source_type=get_source_type())
```

Comment:
`.all()` is redundant because `get_object_or_404` accepts also `Manager` as the argument.

Reply:
Removed `.all()`.